### PR TITLE
Bugfixes + Refactor duplicate code in EmxMetaDataParser

### LINKED ISSUE #5052: Importing will throw a NULL pointer after the second wizard screen
### Reproduce
Select a file to import and press next twice. A message:null will show up

### LINKED ISSUE #5054: Add/Update with the Same EMX will throw an error stating entities are not compatible
### Reproduce
Import TypeTest
Add/Update TypeTest again

### Expected
Import succeeds

### Observed
Import fails stating entities are not compatible

### LINKED ISSUE #5058: Add/Update with packages of the same name will throw an error
#### Reproduce
Upload TypeTest
Add/Update TypeTest again

#### Expected
I can update

#### Observed
Adding existing package [org] is not allowed

### LINKED ISSUE #5061: Entities that are only specified in the attributes sheet do not get their backend set, resulting in a null pointer during importing
#### Reproduce
Import TypeTest
Add/Update TypeTest

#### Expected
Import succeeds

#### Observed
Null pointer

### PR DESCRIPTION
**Pull request by Mark-de-Haan**
- Fix #5052 Importing will throw a NULL pointer after the second wizard screen
- Fix #5054 Add/Update with the Same EMX will throw an error stating entities are not compatible
- Fix #5058 Add/Update with the same package will throw an error
- Fix #5061 Entities that are only specified in the attributes sheet do not get their backend set, resulting in a null pointer during importing

### CODE REVIEW

File: molgenis-data-import/src/main/java/org/molgenis/data/importer/EmxMetaDataParserUtils.java [hunk 1]
```diff
@@ -84,18 +79,33 @@
...
-	private final PackageFactory packageFactory;
-	private final AttributeMetaDataFactory attrMetaFactory;
-	private final EntityMetaDataFactory entityMetaDataFactory;
-
-	public EmxFileOnlyMetaDataParser(PackageFactory packageFactory, AttributeMetaDataFactory attrMetaFactory,
-			EntityMetaDataFactory entityMetaDataFactory)
+	public EmxMetaDataParserUtils(PackageFactory packageFactory, AttributeMetaDataFactory attrMetaFactory,
+			EntityMetaDataFactory entityMetaDataFactory, DataService dataService)
 	{
+		this.dataService = dataService;
```
Comment (bartcharbon) [inline molgenis-data-import/src/main/java/org/molgenis/data/importer/EmxMetaDataParserUtils.java]: require non null

Comment (bartcharbon) [inline molgenis-data-import/src/main/java/org/molgenis/data/importer/EmxMetaDataParserUtils.java]: add explicit constructor without dataService for Gavin.

File: molgenis-data-import/src/main/java/org/molgenis/data/importer/EmxMetaDataParserUtils.java [hunk 2]
```diff
@@ -84,18 +79,33 @@
...
-	private static final String AUTO = "auto";
+	public static final String AUTO = "auto";
 
-	private final PackageFactory packageFactory;
-	private final AttributeMetaDataFactory attrMetaFactory;
-	private final EntityMetaDataFactory entityMetaDataFactory;
-
-	public EmxFileOnlyMetaDataParser(PackageFactory packageFactory, AttributeMetaDataFactory attrMetaFactory,
-			EntityMetaDataFactory entityMetaDataFactory)
+	public EmxMetaDataParserUtils(PackageFactory packageFactory, AttributeMetaDataFactory attrMetaFactory,
```
Comment (bartcharbon) [inline molgenis-data-import/src/main/java/org/molgenis/data/importer/EmxMetaDataParserUtils.java]: not sure what it should be, but I find "Utils" not the most logical name

File: molgenis-data-import/src/main/java/org/molgenis/data/importer/EmxMetaDataParser.java [hunk 3]
```diff
@@ -1,51 +1,59 @@
 package org.molgenis.data.importer;
 
-import com.google.common.collect.*;
+import com.google.common.collect.ImmutableMap;
```
Comment (bartcharbon) [inline molgenis-data-import/src/main/java/org/molgenis/data/importer/EmxMetaDataParser.java]: did you optimize the imports, the list seems a bit odd to me

File: molgenis-data-import/src/main/java/org/molgenis/data/importer/MetaDataParser.java [hunk 4]
```diff
@@ -16,5 +16,10 @@
 	 */
 	ParsedMetaData parse(RepositoryCollection source, String defaultPackage);
 
+	/**
+	 * TODO documentation
```
Comment (bartcharbon) [inline molgenis-data-import/src/main/java/org/molgenis/data/importer/MetaDataParser.java]: please do

File: molgenis-ontology/src/main/java/org/molgenis/ontology/importer/OntologyImportService.java [hunk 5]
```diff
@@ -188,7 +188,7 @@ public boolean getMustChangeEntityName()
 	}
 
 	@Override
-	public LinkedHashMap<String, Boolean> integrationTestMetaData(MetaDataService metaDataService,
+	public LinkedHashMap<String, Boolean> determineImportableEntities(MetaDataService metaDataService,
 			RepositoryCollection repositoryCollection, String defaultPackage)
 	{
 		return metaDataService.integrationTestMetaData(repositoryCollection);
```
Comment (bartcharbon) [inline molgenis-ontology/src/main/java/org/molgenis/ontology/importer/OntologyImportService.java]: maybe rename the metadataservice function as well?

File: molgenis-data-import/src/main/java/org/molgenis/data/importer/EmxMetaDataParser.java (hunk 3 above)
Comment (Mark-de-Haan) [inline molgenis-data-import/src/main/java/org/molgenis/data/importer/EmxMetaDataParser.java]: yes I applied the optimisation. You only use the ImmutableMap from collect package + a handful of static imports so its not necessary to import the entire google.common.collect package

**Pull request merged**